### PR TITLE
Create Game and Author classes

### PR DESCRIPTION
**Pull request by stanleySimeon**
# In this milestone I created:

  - a `Game` class
  - an `Author` class with an association with the Item class
  
  - **methods:**
    `add_item` for the `Game` class
    `can_be_archived` for the `Author` class
   
  - unit tests for all implemented methods
  
  - a `file` contains the following options:
    - list_of_games_and_authors
    - add_game
    
  - a `schema` file contains the schema.sql

### CODE REVIEW

File: author.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Create an Author class with an association to the Item class.
+require './classes/item'
+class Author < Item
```
Comment (myelin0) [inline author.rb]: Author class is not inheriting from Item class according to the diagram given.

File: author.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  attr_accessor :id, :first_name, :last_name, :items
+
+  def initialize(id, first_name, last_name, items)
+    super(first_name, last_name)
```
Comment (myelin0) [inline author.rb]: If you see when you call super (first_name,last_name) those are not properties of super (.ie the Item class), so you can delete this line.

File: game.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'date'
+# create game class
+class Game
```
Comment (myelin0) [inline game.rb]: class Game is inheriting from Item class if you see the diagram. So it should become `class Game< Item` . This means you dont have to rewrite the properties of Item class inside class Game, just use super(publish_date, archived)

File: game.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    @multiplayer = multiplayer
+    @last_played_at = last_played_at
+    @published = published
+    @archived = archived
+  end
+
+  private
+
+  def can_be_archived?
+    super && last_played_at < 2.years.ago
```
Comment (myelin0) [inline game.rb]: last_played_at should be bigger than 2 years, not less according to the instructions.

File: schema/schema.sql [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    published_date DATE,
+    archived boolean,
+    multiplayer boolean,
+    last_played_at DATE,
+);
+
+-- authors table
+CREATE TABLE authors (
+    id INTEGER PRIMARY KEY,
+    name TEXT
```
Comment (myelin0) [inline schema/schema.sql]: should have first_name and last_name VARCHAR(255) NOT NULL (as this class has)

File: schema/schema.sql [hunk 6]
```diff
@@ -0,0 +1,19 @@
+-- games table
+CREATE TABLE IF NOT EXISTS games (
+    id INTEGER PRIMARY KEY,
+    name TEXT,
+    genre TEXT,
+    author TEXT,
```
Comment (myelin0) [inline schema/schema.sql]: the author is a foreign key to this table so you can say: `author_id INTEGER,
    CONSTRAINT fk_author FOREIGN KEY (author_id) REFERENCES author(id) ON DELETE CASCADE`

**Review by myelin0 — CHANGES_REQUESTED** [review]
## Hi @stanleySimeon , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights are:
- [x] No linter errors
- [x] Followed git workflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: author.rb (hunk 1 above)
Comment (stanleySimeon) [reply author.rb]: Your review is so perfect
Great 👍 
Thanks a lot for your help!

File: author.rb (hunk 2 above)
Comment (stanleySimeon) [reply author.rb]: Thanks, I will

File: game.rb (hunk 3 above)
Comment (stanleySimeon) [reply game.rb]: Thanks a lot, I'll do that

File: game.rb (hunk 4 above)
Comment (stanleySimeon) [reply game.rb]: Ooh I'm sorry about that, thanks for your clarification!

File: schema/schema.sql (hunk 6 above)
Comment (stanleySimeon) [reply schema/schema.sql]: Thanks a lot, @myelin0!

File: author.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    @first_name = first_name
+    @last_name = last_name
+    @items = items
+  end
+
+  def add_item(item)
+    @items << item
+    item.author = self
+  end
+end
```
Comment (medsonmoombe) [inline author.rb]: - [x]  Kindly assign the value of instance variable `@items` to equal an empty array as in the diagram and also consider removing and assign it to be randomly generated as shown below
- [ ]  Kindly remove the 'items' argument in the initialize, the `item` is an instance  variable which should be passed in the `def add_item(item)` method .  please kindly implement as demonstrated below
```
  attr_accessor :first_name, :last_name, :item
def initialize(first_name, last_name)
    @id = Random.rand(1..1000)
    @items = []
```
`def add_item(item)`

File: game.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    @multiplayer = multiplayer
+    @last_played_at = last_played_at
+  end
+
+  private
+
+  def can_be_archived?
+    super && last_played_at > 2.years.ago
+  end
+end
```
Comment (medsonmoombe) [inline game.rb]: - [x] The `can_be_archived?` method should return true if parent's method returns true AND if `last_played_at`  is older than 2 years to archive this consider adding the today's date and year as shown below
```
  private

  def can_be_archived?
    super && (Date.today.year - last_played_at.year) > 2
  end
```

File: options/add_game.rb [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+require './game'
+
+def add_game
+  puts 'Add a game:'
+  puts ''
+  puts 'Name:'
+  name = gets.chomp
+  puts 'Multiplayer (y/n):'
+  multiplayer = gets.chomp
+  puts 'Last played at:'
```
Comment (medsonmoombe) [inline options/add_game.rb]: - [x] Kindly add more information to let the user know the kind of data to be entered and the format e.g `puts 'Enter last_played_at(yyyy-mm-dd)'`

**Review by medsonmoombe — CHANGES_REQUESTED** [review]

File: author.rb (hunk 7 above)
Comment (stanleySimeon) [reply author.rb]: Thank you so much, I'll do that

**Review by myelin0 — APPROVED** [review]
Hi @stanleySimeon 

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by medsonmoombe — APPROVED** [review]

**Pull request merged**